### PR TITLE
Custom breadcrumbs

### PR DESCRIPTION
**Pull request by csjh**
### Description

Enables custom handling of breadcrumb naming via frontmatter

Example (from `test-env`):
```md
---
breadcrumb: "SELECT first_name || ' ' || last_name || '''s ' || order_month || ' Order' as breadcrumb FROM orders WHERE id = '{params.order_id}'"
---

Order #{params.order_id}
```
image

### Checklist

- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [x] I have added to the docs where applicable

### CODE REVIEW

**Comment by hughess** [comment]
I think the build error is coming from linking to parent directories that don't have an index page

File: sites/test-env/pages/orders/[order_id].md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+breadcrumb: "SELECT first_name || ' ' || last_name || '''s ' || order_month || ' Order' as breadcrumb FROM orders WHERE id = '{params.order_id}'"
```
Comment (archiewood) [inline sites/test-env/pages/orders/[order_id].md:2]:
```suggestion
breadcrumb: "SELECT first_name || ' ' || last_name || '''s ' || order_month || ' Order' as breadcrumb FROM orders WHERE id = '${params.order_id}'"
```
does this need a `$`

**Comment by archiewood** [comment]
what does this need to get over the line? just QA>

File: sites/example-project/src/pages/+layout.js [hunk 2]
```diff
@@ -120,26 +120,42 @@ export const load = async ({ fetch, route, params, url }) => {
...
+	}
+
+	let tree = pagesManifest;
+	for (const part of (route.id ?? '').split('/').slice(1)) {
+		tree = tree.children[part];
+		if (!tree) break;
+		if (tree.frontMatter?.breadcrumb) {
+			let { breadcrumb } = tree.frontMatter;
+			for (const [param, value] of Object.entries(params)) {
+				breadcrumb = breadcrumb.replaceAll(`{params.${param}}`, value);
```
Comment (archiewood) [inline sites/example-project/src/pages/+layout.js:150]:
```suggestion
				breadcrumb = breadcrumb.replaceAll(`${params.${param}}`, value);
```

Comment (zachstence) [reply sites/example-project/src/pages/+layout.js]: Fixed here [479242a](https://github.com/evidence-dev/evidence/pull/1964/commits/479242a712df2ee60b486c6718eaef94a50480c6)

File: sites/test-env/pages/orders/[order_id].md (hunk 1 above)
Comment (zachstence) [reply sites/test-env/pages/orders/[order_id].md]: Fixed here [479242a](https://github.com/evidence-dev/evidence/pull/1964/commits/479242a712df2ee60b486c6718eaef94a50480c6)

**Comment by zachstence** [comment]
More discussion and changes are necessary here before this is ready to merge

Key issues:
- How do we enable users to specify a custom breadcrumb (string) rather than a dynamic breadcrumb (SQL query)?
- How do we enable users to customize all levels of the breadcrumbs, not just the deepest one?

**Review by ItsMeBrianD — APPROVED** [review]

**Comment by ItsMeBrianD** [comment]
Please make sure that HMR still works as expected

**Pull request merged**